File: 122.Best-Time-to-Buy-and-Sell-Stock-II.py
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/
#
# algorithms
# Easy (50.91%)
# Total Accepted:    293,229
# Total Submissions: 575,993
# beats 100.0% of python submissions

# Greedy，在涨就不卖

class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        length = len(prices)

        if length < 2:
            return 0

        i, res = 0, 0
        while i < length:
            j = i + 1
            while j < length and prices[j] >= prices[j - 1]:
                j += 1
            res += prices[j - 1] - prices[i]
            if j == length:
                break
            i = j

        return res

File: 122.Best-Time-to-Buy-and-Sell-Stock-II.go
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/
//
// algorithms
// Easy (51.83%)
// Total Accepted:    321,838
// Total Submissions: 620,902
// beats 100.0% of golang submissions

package leetcode

func maxProfit(prices []int) int {
	length := len(prices)
	if length < 2 {
		return 0
	}

	var res int

	for i := 1; i < length; i++ {
		if prices[i] > prices[i-1] {
			res += prices[i] - prices[i-1]
		}
	}

	return res
}
